stix2/registration.py

```diff
@@ -83,49 +132,13 @@ def _register_observable(new_observable, version=version.DEFAULT_VERSION):
             None, use latest version.
 
     """
-    properties = new_observable._properties
-
     if not version:
         version = version.DEFAULT_VERSION
 
-    if version == "2.0":
-        # If using STIX2.0, check properties ending in "_ref/s" are ObjectReferenceProperties
-        for prop_name, prop in properties.items():
-            if prop_name.endswith('_ref') and ('ObjectReferenceProperty' not in get_class_hierarchy_names(prop)):
-                raise ValueError(
-                    "'%s' is named like an object reference property but "
-                    "is not an ObjectReferenceProperty." % prop_name,
-                )
-            elif (
-                prop_name.endswith('_refs') and (
-                    'ListProperty' not in get_class_hierarchy_names(prop) or
-                    'ObjectReferenceProperty' not in get_class_hierarchy_names(prop.contained)
-                )
-            ):
-                raise ValueError(
-                    "'%s' is named like an object reference list property but "
-                    "is not a ListProperty containing ObjectReferenceProperty." % prop_name,
-                )
-    else:
-        # If using STIX2.1 (or newer...), check properties ending in "_ref/s" are ReferenceProperties
-        for prop_name, prop in properties.items():
-            if not re.match(PREFIX_21_REGEX, prop_name):
-                raise ValueError("Property name '%s' must begin with an alpha character." % prop_name)
-            elif prop_name.endswith('_ref') and ('ReferenceProperty' not in get_class_hierarchy_names(prop)):
-                raise ValueError(
-                    "'%s' is named like a reference property but "
-                    "is not a ReferenceProperty." % prop_name,
-                )
-            elif (
-                prop_name.endswith('_refs') and (
-                    'ListProperty' not in get_class_hierarchy_names(prop) or
-                    'ReferenceProperty' not in get_class_hierarchy_names(prop.contained)
-                )
-            ):
-                raise ValueError(
-                    "'%s' is named like a reference list property but "
-                    "is not a ListProperty containing ReferenceProperty." % prop_name,
-                )
+    _validate_props(
+        new_observable._properties, version,
+        is_observable20=(version == "2.0"),
+    )
```

Comment:
Alternatively to this approach where `_validate_props()` is responsible for calling `_validate_ref_props()`, we could force each registration helper to call both of those methods themselves, which would  allow this specific one to pass in an alternate `ref_prop_type` kwarg to validate properties against. e.g.:

```python
_validate_props(new_observable._properties, version)
_validate_ref_props(new_observable._properties, version, ref_prop_type=ObjectReferenceProperty)
```

Doing so would introduce the possibility for drift in the validation steps applied between objects again though, which was the original problem with some requirements only being validated for custom observable objects. Not a deal breaker, just a risk to think about.

Reply:
"Reference property validation" is a subset of "property validation", so I like the idea of a general property validation function doing all property validation, as opposed to having special cases it does not validate.  I'm generally good with the design.